packages/jupyter-ai-magics/pyproject.toml

```diff
@@ -50,6 +50,9 @@ all = [
     "ipywidgets",
     "langchain_anthropic",
     "langchain_aws",
+    "cohere!=5.16.0,!=5.16.1",
+    # ^ forbids versions that are incompatible with latest `langchain_cohere`.
+    # Fixed CI in PR #1405.
```

Comment:
`cohere` 5.16 removed `ChatResponse` class that `langchain_cohere` depends on. As of now `langchain_cohere` does not have an upper version bound for `cohere` so will pull in all new `cohere` versions automatically. 

This pin will only disallow cohere 5.16.0 and 5.16.1 and will allow all other future versions. Taking in account that it's improbable that `cohere` will roll back their breaking changes that cause incompatibility with 

`cohere<5.16` pin would work best in this scenario in my opinion and it is what was merged as a part of #1412 into main.

```suggestion
```

Reply:
Here's an issue about it in `langchain_cohere` https://github.com/langchain-ai/langchain-cohere/issues/140